src/Models/DataType.php

```diff
@@ -55,6 +55,31 @@ public function deleteRows()
         return $this->rows()->where('delete', 1);
     }
 
+    public function addRelationshipRow($requestData)
```

Comment:
I feel like it belongs in a controller.  Definitely not in the DataType model though.